cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -2868,8 +2865,8 @@ struct ExtractRegex : public ExtractRegexBase {
       if (input.is_valid && Match(util::string_view(*input.value))) {
         result->value.reserve(group_count);
         for (int i = 0; i < group_count; i++) {
-          result->value.push_back(
-              std::make_shared<ScalarType>(found_values[i].as_string()));
+          result->value.push_back(std::make_shared<ScalarType>(
+              std::make_shared<Buffer>(ToStringView(found_values[i]))));
```

Comment:
This isn't safe, the string_view constructor is non-owning. We need to allocate a buffer/string here.

Reply:
(I suppose it works in tests because RE2 might be returning a slice of the input?)

Reply:
Thanks for catching this.